packages/wallet/backend/src/quote/service.ts

```diff
@@ -109,11 +113,27 @@ export class QuoteService implements IQuoteService {
       params.isReceive &&
       destinationPaymentPointer.assetCode !== asset.code
     ) {
-      const convertedValue = await this.convert({
-        from: assetCode,
-        to: destinationPaymentPointer.assetCode,
-        amount: value
-      })
+      let convertedValue
+      if (isIncomingPayment) {
```

Comment:
I'm not sure about this approach. The users should be able send their desired amounts. If the incoming payment has a receive amount of 100EUR, we should not enforce to send 100EUR.

Reply:
Yep, I agree, the request might be 100EUR but the user ca decide to pay only 50EUR

Reply:
the issue is that converting the value is not exact, so 20.15 euro will result in 100.02 roni (instead of expected 100)
I added a check that if the amount is off by less then 1% to set it to the amount in incoming payment
open to other suggestions if this does not make sense